### DISCUSSION TITLE
Access problem

### DISCUSSION THREAD

**Discussion by zepef** [acheong08/ChatGPT#57] [Q&A]
Hello,

I cannot use you API access. With CLI: 

<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Transitional//EN' 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd'><html xmlns='http://www.w3.org/1999/xhtml'><head><meta content='text/html; charset=utf-8' http-equiv='content-type'/><style type='text/css'>body {font-family:Arial; margin-left:40px; }img  { border:0 none; }#content { margin-left: auto; margin-right: auto }#message h2 { font-size: 20px; font-weight: normal; color: #000000; margin: 34px 0px 0px 0px }#message p  { font-size: 13px; color: #000000; margin: 7px 0px 0px0px}#errorref { font-size: 11px; color: #737373; margin-top: 41px }</style><title>Service unavailable</title></head><body><div id='content'><div id='message'><h2>Our services aren't available right now</h2><p>We're working to restore all services as soon as possible. Please check back soon.</p></div><div id='errorref'><span>0VLuOYwAAAADVkgYPKhpXTZ4ecWbvDBUFSk5CMDJFREdFMDUyMgBlNjZiOGIwMy0wNzkzLTQ0MDktOTc3My0yZTYxMmU3MWExZTM=</span></div></div></body></html>
Something went wrong!
Response is not in the correct format

From Python:

Logging in...
(chatgptapi) PS E:\Documents\LocalRepos\Trials\ChatGPT>  e:; cd 'e:\Documents\LocalRepos\Trials\ChatGPT'; & 'C:\Users\clt\.conda\envs\chatgptapi\python.exe' 'c:\Users\clt\.vscode\extensions\ms-python.python-2022.17.13051012\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher' '54583' '--' 'e:\Documents\LocalRepos\Trials\ChatGPT\chatGPT_test01.py' 
Logging in...
<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Transitional//EN' 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd'><html xmlns='http://www.w3.org/1999/xhtml'><head><meta content='text/html; charset=utf-8' http-equiv='content-type'/><style type='text/css'>body {font-family:Arial; margin-left:40px; }img  { border:0 none; }#content { margin-left: auto; margin-right: auto }#message h2 { font-size: 20px; font-weight: normal; color: #000000; margin: 34px 0px 0px 0px }#message p  { font-size: 13px; color: #000000; margin: 7px 0px 0px0px}#errorref { font-size: 11px; color: #737373; margin-top: 41px }</style><title>Service unavailable</title></head><body><div id='content'><div id='message'><h2>Our services aren't available right now</h2><p>We're working to restore all services as soon as possible. Please check back soon.</p></div><div id='errorref'><span>03bmOYwAAAAAcDmNqI2C0Q5IkbACFZXGOSk5CMDJFREdFMDUwNgBlNjZiOGIwMy0wNzkzLTQ0MDktOTc3My0yZTYxMmU3MWExZTM=</span></div></div></body></html>
Traceback (most recent call last):
  File "C:\Users\clt\.conda\envs\chatgptapi\lib\site-packages\revChatGPT\revChatGPT.py", line 61, in get_chat_text
    response = response.text.splitlines()[-4]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "e:\Documents\LocalRepos\Trials\ChatGPT\chatGPT_test01.py", line 20, in <module>
    resp = chatbot.get_chat_response(prompt, output="text") # Sends a request to the API and returns the response by OpenAI
  File "C:\Users\clt\.conda\envs\chatgptapi\lib\site-packages\revChatGPT\revChatGPT.py", line 85, in get_chat_response
    return self.get_chat_text(data)
  File "C:\Users\clt\.conda\envs\chatgptapi\lib\site-packages\revChatGPT\revChatGPT.py", line 65, in get_chat_text
    raise ValueError("Response is not in the correct format")
ValueError: Response is not in the correct format

What am I doing wrong?

Thanks for the good work!!!

**Comment by acheong08** [comment]
It's back up.

**Reply by zepef** [reply]
I'ill try it right now THANKS!

**Reply by zepef** [reply]
Now I get:

Logging in...
{"detail":{"message":"Incorrect API key provided: Bearer. You can find your API key at https://beta.openai.com.","type":"invalid_request_error","param":null,"code":"invalid_api_key"}}
Traceback (most recent call last):
  File "C:\Users\clt\.conda\envs\chatgptapi\lib\site-packages\revChatGPT\revChatGPT.py", line 61, in get_chat_text
    response = response.text.splitlines()[-4]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "e:\Documents\LocalRepos\Trials\ChatGPT\chatGPT_test01.py", line 20, in <module>
    resp = chatbot.get_chat_response(prompt, output="text") # Sends a request to the API and returns the response by OpenAI
  File "C:\Users\clt\.conda\envs\chatgptapi\lib\site-packages\revChatGPT\revChatGPT.py", line 85, in get_chat_response
    return self.get_chat_text(data)
  File "C:\Users\clt\.conda\envs\chatgptapi\lib\site-packages\revChatGPT\revChatGPT.py", line 65, in get_chat_text
    raise ValueError("Response is not in the correct format")
ValueError: Response is not in the correct format

Even if my Open AI key is OK as well as my session token is fresh

It works fine from command line...

**Reply by acheong08** [reply]
They must have changed the API. Fixing

**Reply by zepef** [reply]
It works great now, I was getting it wrong with the API key / Password. I just remplaced the JSON pair in your example Python code with the AUTH/SESSION pair from config.json. I feel stupid ;-)
I still have to do a little formating for the response.
Thanks a lot again !

**Reply by acheong08** [reply]
I am slightly confused. Were you using email/password? Email/password should work. If it doesn't can you inform me?

**Reply by zepef** [reply]
In fact it was returning an **Incorrect API key provided: Bearer** error even when I rotated my password. I had to use the old OAUTH way. Now I am getting problems with: **NameError: name 'clear_session_cookies' is not defined** from VSCODE.
And from command line:

Press enter twice to submit your question.

Error refreshing session
<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Transitional//EN' 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd'><html xmlns='http://www.w3.org/1999/xhtml'><head><meta content='text/html; charset=utf-8' http-equiv='content-type'/><style type='text/css'>body { font-family:Arial; margin-left:40px; }img  { border:0 none; }#content { margin-left: auto; margin-right: auto }#message h2 { font-size: 20px; font-weight: normal; color: #000000; margin: 34px 0px 0px 0px }#message p  { font-size: 13px; color: #000000; margin: 7px 0px 0px 0px }#errorref { font-size: 11px; color: #737373; margin-top: 41px }</style><title>Microsoft</title></head><body><div id='content'><div id='message'><h2>The request is blocked.</h2></div><div id='errorref'><span>0gM6OYwAAAADw8ZZzqJA3QauWcOZxiDsCSk5CMDJFREdFMDUxOQBlNjZiOGIwMy0wNzkzLTQ0MDktOTc3My0yZTYxMmU3MWExZTM=</span></div></div></body></html>

You:
what is a chatbot?

Chatbot:
<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Transitional//EN' 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd'><html xmlns='http://www.w3.org/1999/xhtml'><head><meta content='text/html; charset=utf-8' http-equiv='content-type'/><style type='text/css'>body { font-family:Arial; margin-left:40px; }img  { border:0 none; }#content { margin-left: auto; margin-right: auto }#message h2 { font-size: 20px; font-weight: normal; color: #000000; margin: 34px 0px 0px 0px }#message p  { font-size: 13px; color: #000000; margin: 7px 0px 0px 0px }#errorref { font-size: 11px; color: #737373; margin-top: 41px }</style><title>Microsoft</title></head><body><div id='content'><div id='message'><h2>The request is blocked.</h2></div><div id='errorref'><span>0k86OYwAAAADq3fbmeimGQL2HPh8TNCPVSk5CMDJFREdFMDUwOABlNjZiOGIwMy0wNzkzLTQ0MDktOTc3My0yZTYxMmU3MWExZTM=</span></div></div></body></html>
Something went wrong!
Response is not in the correct format

Maybe they are trying to block access?

I'll keep you updated
All the best!

**Reply by acheong08** [reply]
Check out the new readme. This library just got killed by OpenAI

**Reply by zepef** [reply]
That is really too bad your lib was good! Maybe they are building their own?
Anyway, bravo again!

**Reply by zepef** [reply]
Now they have access problems even on GPT-3/Da Vinci 003...

**Reply by acheong08** [reply]
Updated Readme. It has been fixed

**Reply by acheong08** [reply]
I went into panic mode and found the issue

**Reply by zepef** [reply]
Panic is Good! It works great! I have to fix the formatting of the generator output. My prompt/completion pair is long and at completion the for loop is giving me all iterations of the text.
I'll keep you updated.
All the best!

**Reply by zepef** [reply]
I just used "text" instead of "sream" to get final completion
Thanks again!
